src/stories/containers/EndgameBudgetContainerSecondLevel/useEndgameBudgetContainerSecondLevel.tsx

```diff
@@ -43,20 +43,31 @@ export const useEndgameBudgetContainerSecondLevel = (budgets: Budget[]) => {
     handlePeriodChange,
     periodicSelectionFilter,
     handleSelectChangeMetrics,
-  } = useBreakdownTable();
+  } = useBreakdownTable(initialYear);
 
   const { headersExpenseReport, reportExpenseItems, handleLoadMore, showSome, onSortClick } =
     useDelegateExpenseTrendFinances();
 
-  const levelBudget = budgets?.find((budget) => budget.codePath === levelCode);
+  const levelBudget = budgets.find((budget) => budget.codePath === levelPath);
   const title = removePrefix(levelBudget?.name || '', prefixToRemove) || '';
   const { isLight } = useThemeContext();
-  const [year, setYears] = useState<string>('2022');
+  const [year, setYear] = useState(initialYear);
   const isMobile = useMediaQuery(lightTheme.breakpoints.down('tablet_768'));
   const icon = levelBudget?.image || '';
   // const icon = itemTitle?.image || '';
   const handleChangeYearsEndgameAtlasBudget = (value: string) => {
-    setYears(value);
+    setYear(value);
+    router.push(
+      {
+        pathname: '/finances/[firstPath]',
+        query: {
+          firstPath: router.query.firstPath,
+          year: value,
+        },
+      }
+      /* undefined,
+      { shallow: true } */
```

Comment:
The third level is using shallow navigation, should not have the same behaviorn on both pages?

Reply:
When we start to handle analytics-related data with SWR, all levels will perform shallow routing. That's why I left the commented code there.